platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/TestEventGenerator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.testing;
+
+import org.gradle.api.Incubating;
+
+import java.time.Instant;
+
+/**
+ * Generates test events.
+ *
+ * @since 8.12
+ */
+@Incubating
+public interface TestEventGenerator extends AutoCloseable {
```

Comment:
`Reporter` or maybe `Sender` I would agree with. I don't know that we'll always record the events so `Recorder` might be wrong, and `Logger` feels wrong because it implies some amount of printing to me?